contentcuration/contentcuration/frontend/shared/data/serverSync.js

```diff
@@ -331,6 +331,8 @@ const debouncedSyncChanges = debounce(() => {
   if (!syncActive) {
     return syncChanges();
   }
+  // TODO: actually return promise that resolves when active sync completes
+  return new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
We may be able to use the p-debounce library for this in future, or vendor and modify it like I have here: https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/composables/useProgressTracking.js#L474

Reply:
Prathamesh and I have been integrating streams into this sync code for websockets, which eliminates this `syncActive` state management and will make it easier to determine when it's complete.